### ISSUE TITLE
[Feature Request] Tinting foreground and background colours

### ISSUE THREAD

**Issue by davepinto** [dylanaraps/pywal#254]
I'd like to request a feature that allows the tinting of the background color, and using the tints in a template file. This could be used for example, to darken, or lighten the cursorline in vim, easing visibility whilst keeping good aesthetics.

**Comment by dylanaraps** [comment]
By tinting do you mean changing the brightness values of both?

**Comment by dylanaraps** [comment]
Would a command-line flag suffice? Something like `wal -i img.jpg --bg-tint '20%' --fg-tint '20%'`.

**Comment by davepinto** [comment]
I don't think so, because in order for this to work, I need the normal background color, and then a tinted background color together. Here's a screenshot of what I mean: https://imgur.com/a/zK6tExu
On the left is how it usually shows up when I set ctermbg to 0, on the right is what I want. For example I want to be able to do something like this:
`*color8: {background20}`
Which would give me something like the background color but 30 percent darker (assuming background50 is default).

**Comment by dylanaraps** [comment]
I see, I'll work on this. :+1:

**Comment by dylanaraps** [comment]
I don't think you can set the value of properties from the format strings (`{color0.lighten(20)}`). We could dynamically generate 100 properties for the `Color` object at runtime to get around this but it's kinda ugly.

I'll continue working on this.

**Comment by dylanaraps** [comment]
An alternative is looking into a more powerful templating method which will break backwards compatibility.

**Comment by davepinto** [comment]
I can work on this if you'd like. I feel like the current templating method is a little too rigid. For example, another problem I ran into was trying to use the alpha value to change the transparency of the polybar. Polybar takes an ARGB hex value for the colour, but the template files only provide the `var.xrgba` value, forcing me to create my own argb function in util.py.

**Comment by davepinto** [comment]
#258 
I've done a thing. Sorry I've taken so long, I haven't had much time to work on extra projects lately. Too much study. Also, I'm still very new to github, so I'm not sure if I've done everything right. If there's anything I've messed up, or you need me to change, please tell me.

**Comment by dylanaraps** [comment]
I’ll take a look at this when I get home. Cheers for working on it. 👍

**Comment by dustinlacewell** [comment]
I strongly believe we should adopt Jinja2 which would allow for extremely expressive templates. Passing values to functions and so on would be a breeze.
